### PR TITLE
Clean up Event Filter (from #5)

### PR DESCRIPTION
**Pull request by 313ctric**
Fixes various cleanup issues in #5.

- [X] Use bootstraps outline button with custom colours instead of custom hovers
- [X] Add a small top margin to the Tags header
- [X] Added "Other" filter
- [X] Move the JS script into file in macros and include it in the template

I didn't put the filters in the sidebar as this would mean they are at the bottom of the page on mobile, which makes them pretty useless there.

I also cleaned up some unclosed `<div>`s in the `term()` macro and made them only show for the grid layout as otherwise this renders lots of empty space when filters are used.

### CODE REVIEW

File: templates/macros/events.html [hunk 1]
```diff
@@ -314,4 +317,265 @@ <h2 class="text-center">No events {% if archive %}finished{% endif %} yet, pleas
...
+
+{% macro filter_script() %}
+<script>
+  // these 'categories' are actually specified by the 'colour'
+  // property, but they fit well enough to categorise events
+  // and are probably more useful than the tags
+  // taken from colours.json
+  const ALLOWED_CATEGORIES = [
+      "gaming", "academic", "tech", "social", "inclusivity",
+      "publicity", "compcafe", "milk", "other"
```
Comment (Mole1424) [inline templates/macros/events.html]: other category never seems to appear?
publicity also don't have a specifc event associated with them

File: templates/macros/events.html [hunk 2]
```diff
@@ -314,4 +317,265 @@ <h2 class="text-center">No events {% if archive %}finished{% endif %} yet, pleas
...
+      "compcafe":   '{{- formatters::colour(colour="compcafe")    -}}',
+      "other":      '{{- formatters::colour(colour="secretary")   -}}',
+  };
+  const CATEGORY_DISPLAY_NAMES = {
+      "gaming":     'Gaming',
+      "academic":   'Academic',
+      "tech":       'Tech',
+      "social":     'Social',
+      "inclusivity":'Inclusivity',
+      "publicity":  'Publicity',
```
Comment (Mole1424) [inline templates/macros/events.html]: see above comment on publicity

**Review by Mole1424 — COMMENTED** [review]
main thing dont need publicity

also potential arguement for comp cafe to be merged with academic?

**Comment by 313ctric** [comment]
I've removed publiicity and compcafe as nobody uses them.

The other category only shows when an event doesn't fit in any of the other categories, and as all the events currently in the system fit in one of the other categories it never shows for the current set of events.
If you locally change an event to have a category that doesn't have a specific filter then it will appear in the `other` category.

**Pull request merged**